### PR TITLE
FrontFlow/Red: New Package.

### PR DESCRIPTION
**Pull request by tkameyama**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ffr/package.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    for Industrial Science(FSIS) project supported by IT program of
+    Ministry of Education, Culture, Sports, Science and Technology(MEXT),
+    JAPAN."""
+
+    homepage = "http://www.ciss.iis.u-tokyo.ac.jp/rss21/theme/multi/fluid/fluid_softwareinfo.html"
+
+    version('3.1.004', sha256='2b396f66bb6437366721fac987f9c6e8b830638c3e4cb5df6a08ff41633f8481', url="file://{0}/FFR_V3.1.004.zip".format(os.getcwd()))
+    version('3.0_000', sha256='edc69fb1fd9dbdb3f531a8f2b9533a9b3c1a28768bb4029b84a6b35c95db0b48', url="file://{0}/open_FrontFlowRed_3.0_000.tar.gz".format(os.getcwd()))
+
+    # FronntFroow/red is used Fortran format I/E without width (Foe Example 3I)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ffr/package.py:21]:
```suggestion
    # FrontFlow/red uses Fortran format I/E without width (For Example 3I)
```

File: var/spack/repos/builtin/packages/ffr/package.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                make = join_path(workdir, 'Makefile')
+                os.chmod(make, 0o644)
+                m = FileFilter(make)
+                m.filter(
+                    r'include Makefile\..*\.in',
+                    'include Makefile.spack.in'
+                )
+                m.filter(
+                    'LIBPRE =.*$',
+                    'LIBPRE = -L{0} -lmetis'.format(spec['metis'].prefix.lib)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ffr/package.py:61]:
```suggestion
                    'LIBPRE = ' + spec['metis'].libs.ld_flags
```
This has the advantage that it works regardless of whether the libraries are found in `prefix.lib` or `prefix.lib64`.

File: var/spack/repos/builtin/packages/ffr/package.py (hunk 1 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/ffr/package.py]: fixed

File: var/spack/repos/builtin/packages/ffr/package.py (hunk 2 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/ffr/package.py]: fixed

File: var/spack/repos/builtin/packages/ffr/package.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    for Industrial Science(FSIS) project supported by IT program of
+    Ministry of Education, Culture, Sports, Science and Technology(MEXT),
+    JAPAN."""
+
+    homepage = "http://www.ciss.iis.u-tokyo.ac.jp/rss21/theme/multi/fluid/fluid_softwareinfo.html"
+
+    version('3.1.004', sha256='2b396f66bb6437366721fac987f9c6e8b830638c3e4cb5df6a08ff41633f8481', url="file://{0}/FFR_V3.1.004.zip".format(os.getcwd()))
+    version('3.0_000', sha256='edc69fb1fd9dbdb3f531a8f2b9533a9b3c1a28768bb4029b84a6b35c95db0b48', url="file://{0}/open_FrontFlowRed_3.0_000.tar.gz".format(os.getcwd()))
+
+    # FronntFroow/red used Fortran format I/E without width (For Example 3I)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ffr/package.py:21]:
```suggestion
    # FrontFlow/red used Fortran format I/E without width (For Example 3I)
```

File: var/spack/repos/builtin/packages/ffr/package.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    for Industrial Science(FSIS) project supported by IT program of
+    Ministry of Education, Culture, Sports, Science and Technology(MEXT),
+    JAPAN."""
+
+    homepage = "http://www.ciss.iis.u-tokyo.ac.jp/rss21/theme/multi/fluid/fluid_softwareinfo.html"
+
+    version('3.1.004', sha256='2b396f66bb6437366721fac987f9c6e8b830638c3e4cb5df6a08ff41633f8481', url="file://{0}/FFR_V3.1.004.zip".format(os.getcwd()))
+    version('3.0_000', sha256='edc69fb1fd9dbdb3f531a8f2b9533a9b3c1a28768bb4029b84a6b35c95db0b48', url="file://{0}/open_FrontFlowRed_3.0_000.tar.gz".format(os.getcwd()))
+
+    # FronntFlow/red used Fortran format I/E without width (For Example 3I)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ffr/package.py:21]:
```suggestion
    # FrontFlow/red used Fortran format I/E without width (For Example 3I)
```

**Pull request merged**